functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -6,71 +10,124 @@
+            logging.debug('UPDATE query executed')
+            logging.info(f"Executed the query: {rowcount} rows affected for taskId {param2}")
+            return func.HttpResponse(status_code=200)
+        except Exception as e:
+            logging.critical('Unable to execute the query')
+            return func.HttpResponse("Error: %s" % str(e), status_code=400)
+    finally:
+        #commits changes to db
         cnxn.commit()
-        return rowcount
-#irrespective of results, closes connection
-    finally: 
+        #properly closes the connection
         cursor.close()
         cnxn.close()
-        logging.info('Closed the connection')
+        logging.info('Closed the db connection')    
 
 #DELETE API method function
-def delete(param):
-    return ('You selected POST method with {param} values. Functionality under construction')
-            # STARTER CODE FOR NEXT SPRINT, SQL QUERY TESTED
-            #     # #delete row, client passes in userId and taskId
-            #     sql_query = ("""DELETE FROM [dbo].[tasks] WHERE userId = ? AND taskId = ?""")
-            #     cursor.execute(sql_query, userId, taskId)
-            #     logging.info('Executed the query')
-            #     cnxn.commit()
+def delete(param1, param2):
+    #connects to db
+    cnxn = connect()
+    cursor = cnxn.cursor()
+    try:
+        logging.info('opened connection')        
+        logging.info('Going to execute DELETE task query')
+        # # task_req_body input check
+        # logging.debug("Checking the request body for necessary fields to update a task")
+        # try:
+        #     assert 'title' in task_req_body, "New task request body did not contain field: 'title'"
+        #     assert 'description' in task_req_body, "New task request body did not contain field: 'description'"
+        #     assert 'dueDate' in task_req_body, "New user request body did not contain field: 'dueDate'"
+        # except AssertionError as task_req_body_content_error:
+        #     logging.critical('New user request body did not contain fields to update a task')
+        #     return func.HttpResponse(task_req_body_content_error.args[0], status_code=400)
+        # logging.info('New user request body contained necessary fields to update a task')  
+        # # Unpack task data
+        # title = task_req_body.get('title')
+        # description = task_req_body.get('description')
+        # dueDate = datetime.strptime(task_req_body.get('dueDate'), '%d/%m/%y %H:%M:%S')     
+
+        # #delete row, client passes in userId and taskId
+        sql_query = ("""DELETE FROM [dbo].[tasks] WHERE userId = ? AND taskId = ?""")
+        try:
+            rowcount = cursor.execute(sql_query, param1, param2).rowcount
+            if not rowcount:
+                logging.error('No record with the requested parameters exists in db')
+                return func.HttpResponse('No record to delete', status_code=500)                

```

Comment:
I think this is probably supposed to be a 404 response, because the resource was not found

Reply:
My approach is that as we are not going to have the end user working with this endpoint, the check for the appropriate param should be happening on the client side, which would indicate their software is not working properly / raise 500 error code. If not 500, I'd use 400 "Bad Request" and steer away from 404 in this instance.